staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -177,6 +177,32 @@ type Backoff struct {
 	Factor   float64       // Duration is multiplied by factor each iteration
 	Jitter   float64       // The amount of jitter applied each iteration
 	Steps    int           // Exit with error after this many steps
+	Cap      time.Duration // The maximum amount of time to wait if set
+}
+
+// Step returns the next interval in the exponential backoff.
+func (b *Backoff) Step() time.Duration {
```

Comment:
I know it's annoying, but can you move this refactor to separate PR?

Reply:
I will once we tease apart the retry design.